pkg/cloud/aws/actuators/machine/actuator.go

```diff
@@ -59,6 +61,51 @@ func NewActuator(params ActuatorParams) *Actuator {
 	}
 }
 
+func (a *Actuator) getClusterMachines(ms *actuators.MachineScope, cluster *clusterv1.Cluster) (*clusterv1.MachineList, error) {
+	clusterMachines, err := ms.MachineClient.List(v1.ListOptions{})
+	if err != nil {
+		return nil, fmt.Errorf("failed to get machinesList for cluster %q in namespace %q: %v", cluster.Name, cluster.Namespace, err)
+	}
+	return clusterMachines, nil
+}
+
+func (a *Actuator) getControlPlaneMachines(machineList *clusterv1.MachineList) []*clusterv1.Machine {
+	var cpm []*clusterv1.Machine
+	for _, m := range machineList.Items {
+		if m.Spec.Versions.ControlPlane != "" {
+			cpm = append(cpm, m.DeepCopy())
+		}
+	}
+	return cpm
+}
+
+func isMachinesSame(m1 *clusterv1.Machine, m2 *clusterv1.Machine) bool {
```

Comment:
Rename this to `machinesEqual`.

Add a documentation note that we are defining equality as name and namespace are equivalent and that we are not checking any other fields.